src/envoy/mixer/http_filter.cc

```diff
@@ -149,6 +185,24 @@ class Instance : public Http::StreamFilter, public Http::AccessLog::Instance {
   FilterHeadersStatus decodeHeaders(HeaderMap& headers,
                                     bool end_stream) override {
     Log().debug("Called Mixer::Instance : {}", __func__);
+
+    // forward attributes (off by default)
+    if (!config_->forward_attributes().empty()) {
+      // detect disable switch in the opaque config
+      auto route = decoder_callbacks_->route()->routeEntry();
```

Comment:
We may need to optimize this. We don't want to call this for each request.

 Maybe create a map<route, bool>  in the Config to cache the result.
Or two maps, one for mixer_control and one for mixer_forward.